### PR TITLE
Pre-Merge Schema validation

### PR DESCRIPTION
**Pull request by kdelee**
##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

- Continuous Integration

##### AWX VERSION

```
n/a
```

##### ADDITIONAL INFORMATION

Related to: https://github.com/ansible/awx/pull/2270

@Spredzy this defines the check, need to make non-voting
Also need to figure out what is needed to allow replacement of reference schema. Will work on that locally and let you know whats up. We can at least get the non-voting check running.

### CODE REVIEW

**Comment by kdelee** [comment]
You can run locally with `tox -e validate-schema -- validate-schema`
Or if you just want to run the make command and not build the container `make docker-compose-validate-schema `

**Comment by kdelee** [comment]
Looks like upon merge/acceptance of any schema changes we'll want to do the following:
(assuming presence of `aws` cli client and that it is configured to have access to the s3 bucket https://s3.amazonaws.com/awx-public-ci-files )
```
make docker-compose-genschema
aws s3 cp ./schema.json s3://awx-public-ci-files/schema.json --acl public-read
```

Added value is that we can add to CONTRIBUTING.md that folks will be able to point https://editor.swagger.io/ at https://s3.amazonaws.com/awx-public-ci-files/schema.json to explore the api
@kialam ^

**Comment by kdelee** [comment]
hrm

```
2018-11-15 21:29:58.875327 | static | api-lint runtests: commands[0] | flake8
2018-11-15 21:29:58.875391 | static | setting PATH=/home/zuul/src/github.com/ansible/awx/.tox/api-lint/bin:/usr/local/bin:/usr/bin
2018-11-15 21:29:58.878098 | static |   /home/zuul/src/github.com/ansible/awx$ /home/zuul/src/github.com/ansible/awx/.tox/api-lint/bin/flake8
2018-11-15 21:30:11.147364 | static | ./awx/main/tests/docs/test_swagger_generation.py:175:15: W605 invalid escape sequence '\s'
2018-11-15 21:30:11.147528 | static | ./awx/main/tests/docs/test_swagger_generation.py:176:42: W605 invalid escape sequence '\,'
```

**Review by ryanpetrello — APPROVED** [review]
This is awesome 😄

File: awx/main/tests/docs/test_swagger_generation.py [hunk 1]
```diff
@@ -168,4 +172,9 @@ def teardown_class(cls):
                 '2018-02-01T08:00:00.000000Z',
                 data
             )
+            data = re.sub( # noqa W605
+                '''(\s+"client_id": ")([a-zA-Z0-9]{40})("\,\s*)''', # noqa W605
```
Comment (ryanpetrello) [inline awx/main/tests/docs/test_swagger_generation.py]: `r'''(\s+"client_id": ")([a-zA-Z0-9]{40})("\,\s*)'''`

File: awx/main/tests/docs/test_swagger_generation.py [hunk 2]
```diff
@@ -168,4 +172,9 @@ def teardown_class(cls):
...
+                '\\1xxxx\\3',
```
Comment (ryanpetrello) [inline awx/main/tests/docs/test_swagger_generation.py]: Same here, start w/ an `r` literal

File: awx/main/tests/docs/test_swagger_generation.py (hunk 1 above)
Comment (kdelee) [reply awx/main/tests/docs/test_swagger_generation.py]: ah cool

**Comment by Spredzy** [comment]
@kdelee  It works - it already spotted a (non code related) issue (curl retrieved content has a new line at end of file which `schema.json` does not:

```
diff -u schema.json reference-schema.json                                                                                                                                                     
--- schema.json 2018-11-16 10:12:57.600501776 +0100                                                                                                                                           
+++ reference-schema.json       2018-11-16 10:13:01.279420809 +0100
@@ -24630,4 +24630,4 @@
   "consumes": [
     "application/json"
   ]
-}
\ No newline at end of file
+}
```

Adding the following line in the Makefile solves it:

```patch
diff --git a/Makefile b/Makefile
index d4857a12a..015ab18b2 100644
--- a/Makefile
+++ b/Makefile
@@ -578,6 +578,7 @@ docker-compose-genschema:
 docker-compose-validate-schema:
        $(MAKE) docker-compose-genschema
        curl https://s3.amazonaws.com/awx-public-ci-files/schema.json -o reference-schema.json
+       truncate -s -1 reference-schema.json
        diff -u schema.json reference-schema.json
 
 docker-compose-clean:
```

**Review by Spredzy — CHANGES_REQUESTED** [review]
Small change required else looks great and is working.

**Comment by kdelee** [comment]
> ```
> Adding the following line in the Makefile solves it:
> 
> ```diff
> diff --git a/Makefile b/Makefile
> index d4857a12a..015ab18b2 100644
> --- a/Makefile
> +++ b/Makefile
> @@ -578,6 +578,7 @@ docker-compose-genschema:
>  docker-compose-validate-schema:
>         $(MAKE) docker-compose-genschema
>         curl https://s3.amazonaws.com/awx-public-ci-files/schema.json -o reference-schema.json
> +       truncate -s -1 reference-schema.json
>         diff -u schema.json reference-schema.json
>  
>  docker-compose-clean:
> ```

I wonder instead about some of these `diff` options:
```
       -Z, --ignore-trailing-space
              ignore white space at line end

       -b, --ignore-space-change
              ignore changes in the amount of white space

       -w, --ignore-all-space
              ignore all white space

       -B, --ignore-blank-lines
              ignore changes where lines are all blank

```

**Comment by kdelee** [comment]
```
make[1]: Leaving directory `/awx_devel'
mv swagger.json schema.json
make[1]: Leaving directory '/home/elijah/sfw/ansible/kdelee_awx'
curl https://s3.amazonaws.com/awx-public-ci-files/schema.json -o reference-schema.json
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1572k  100 1572k    0     0  2430k      0 --:--:-- --:--:-- --:--:-- 2426k
# Ignore differences in whitespace with -b
diff -u -b schema.json reference-schema.json

elijah > echo $?
0

```

**Review by Spredzy — APPROVED** [review]
LGTM

**Review by matburt — APPROVED** [review]

**Pull request merged**